### ISSUE TITLE
Legion hangs right before returning, when using Spy

### ISSUE THREAD

**Issue by manopapad** [StanfordLegion/legion#416]
I can reproduce this consistently on sapling, running on Soleil master (commit de0aeea04). Compiling Legion with Spy seems necessary to reproduce this.

Follow the instructions at https://github.com/stanfordhpccenter/soleil-x to set up Soleil on sapling. Make sure you compile Legion with `DEBUG=1 USE_SPY=1`. Afterwards, run Soleil as follows:

```
./soleil.sh -m ../testcases/hit_to_openchannel.dbg.json -logfile spy_%.log
```

Some relevant comments from people in the know, while trying to debug this:
* Task `workDual` has 197 outstanding deletions in it.
* An untriggered Realm event `8000000047500009` is produced. It's not a user event because it doesn't show up in the Spy logs.

Legion commit 97eaf79f was expected to fix this, but didn't.

**Comment by lightsighter** [comment]
Actually it is an untriggered user event. Probably occurring due to the hacks that we have in place to help Legion Spy understand how fence (or in this case fence-like deletion) operations work. It should only occur if you compile with -DLEGION_SPY. If anyone observes the behavior not in this mode then please let us know.

**Comment by lightsighter** [comment]
@manopapad Can we still reproduce this or is it fixed?

**Comment by manopapad** [comment]
I can still reproduce this on sapling

**Comment by lightsighter** [comment]
Can you get me processes on sapling for this running with debug mode and -ll:force_kthreads in the nopaint branch?

**Comment by lightsighter** [comment]
Fixed with f09424c
